cGame_logic.cpp

```diff
@@ -1698,18 +1667,94 @@ void cGame::setNextStateToTransitionTo(int newState) {
 
 void cGame::drawCombatMouse() {
     if (mouse->isBoxSelecting()) {
-        allegroDrawer->drawRectangle(bmp_screen, mouse->getBoxSelectRectangle(), game.getColorFadeSelected(255, 255, 255));
+        allegroDrawer->drawRectangle(bmp_screen, mouse->getBoxSelectRectangle(),
+                                     game.getColorFadeSelected(255, 255, 255));
     }
 
-    if(mouse->isMapScrolling()) {
+    if (mouse->isMapScrolling()) {
         cPoint startPoint = mouse->getDragLineStartPoint();
         cPoint endPoint = mouse->getDragLineEndPoint();
-        allegroDrawer->drawLine(bmp_screen, startPoint.x, startPoint.y, endPoint.x, endPoint.y, game.getColorFadeSelected(255, 255, 255));
+        allegroDrawer->drawLine(bmp_screen, startPoint.x, startPoint.y, endPoint.x, endPoint.y,
+                                game.getColorFadeSelected(255, 255, 255));
     }
 
     mouse->draw();
 }
 
-void cGame::onCombatMouseEventMovedTo(const s_MouseEvent &event) {
-    // not needed?
+void cGame::takeScreenshot() {
+    char filename[25];
```

Comment:
Ideally, we'll use the std::string type, as that has no upper bound. But I remember an issue where lib fmt is considered. That's probably a better idea. We can convert everything then.

Reply:
In this case I simply moved a function, so I did not really bother rewriting it. But, by all means, I do want to learn/improve. So if you have a (different/better) implementation for this (via a small PR?), I am very interested.